uber/models.py

```diff
@@ -494,93 +494,138 @@ def lookup_attendee(self, full_name, email, zip_code):
                     return attendee[0]
             raise ValueError('attendee not found')
 
-        def next_badge_num(self, badge_type, old_badge_num):
-            # assert_badge_locked()
-            badge_type = int(badge_type)
+        def needs_badge_num(self, attendee=None, badge_type=None):
```

Comment:
This is the first part of the refactor. Centralizing this logic means that it can now be overridden in a plugin, should an event choose to do so.